engine/consensus/approvals/assignment_collector_statemachine.go

```diff
@@ -0,0 +1,264 @@
+	collector := asm.atomicLoadCollector()
+	return collector.CheckEmergencySealing(observer, finalizedBlockHeight)
+}
+
+func (asm *AssignmentCollectorStateMachine) RequestMissingApprovals(observer consensus.SealingObservation, maxHeightForRequesting uint64) (uint, error) {
+	collector := asm.atomicLoadCollector()
+	return collector.RequestMissingApprovals(observer, maxHeightForRequesting)
+}
+
+func (asm *AssignmentCollectorStateMachine) ProcessingStatus() ProcessingStatus {
+	collector := asm.atomicLoadCollector()
+	return collector.ProcessingStatus()
+}
+
+// ChangeProcessingStatus changes the AssignmentCollector's internal processing
+// status. The operation is implemented as an atomic compare-and-swap, i.e. the
+// state transition is only executed if AssignmentCollector's internal state is
+// equal to `expectedValue`. The return indicates whether the state was updated.
+// The implementation only allows the transitions
+//         CachingApprovals   -> VerifyingApprovals
+//         CachingApprovals   -> Orphaned
+//    and  VerifyingApprovals -> Orphaned
+// Error returns:
+// * nil if the state transition was successfully executed
+// * ErrDifferentCollectorState if the AssignmentCollector's state is different than expectedCurrentStatus
+// * ErrInvalidCollectorStateTransition if the given state transition is impossible
+// * all other errors are unexpected and potential symptoms of internal bugs or state corruption (fatal)
+func (asm *AssignmentCollectorStateMachine) ChangeProcessingStatus(expectedCurrentStatus, newStatus ProcessingStatus) error {
+	// don't transition between same states
+	if expectedCurrentStatus == newStatus {
+		return nil
+	}
+
+	// state transition: VerifyingApprovals -> Orphaned
+	if (expectedCurrentStatus == VerifyingApprovals) && (newStatus == Orphaned) {
+		_, err := asm.verifying2Orphaned()
+		if err != nil {
+			return fmt.Errorf("failed to transistion AssignmentCollector from %s to %s: %w", expectedCurrentStatus.String(), newStatus.String(), err)
+		}
+		return nil
+	}
+
+	// state transition: CachingApprovals -> Orphaned
+	if (expectedCurrentStatus == CachingApprovals) && (newStatus == Orphaned) {
+		_, err := asm.caching2Orphaned()
+		if err != nil {
+			return fmt.Errorf("failed to transistion AssignmentCollector from %s to %s: %w", expectedCurrentStatus.String(), newStatus.String(), err)
+		}
+		return nil
+	}
+
+	// state transition: CachingApprovals -> VerifyingApprovals
+	if (expectedCurrentStatus == CachingApprovals) && (newStatus == VerifyingApprovals) {
+		cachingCollector, err := asm.caching2Verifying()
+		if err != nil {
+			return fmt.Errorf("failed to transistion AssignmentCollector from %s to %s: %w", expectedCurrentStatus.String(), newStatus.String(), err)
+		}
+
+		// re-ingest IncorporatedResults and Approvals that were stored in the cachingCollector
```

Comment:
```suggestion
		// let the Verifying collector to re-process IncorporatedResults and Approvals that were stored 
		// in the cachingCollector.
		// if meanwhile there are other incorporated results or approvals added to the caching collector,
		// there are checks to ensure they will be reprocessed again. By that time, it would be the verifying
		// collector to process them.
```